packages/server/src/instances/service.ts

```diff
@@ -117,14 +120,16 @@ export class InstanceService extends Service {
   }
 
   async monitor() {
-    await this.monitoring.setup()
+    await this.monitoring.monitor()
   }
 
   async initialize(conduitId: uuid) {
     const instance = await this.get(conduitId)
 
     try {
-      await instance.initialize()
+      await this.distributedService.using(`lock_dyn_instance_init::${conduitId}`, async () => {
+        await instance.initialize()
+      })
```

Comment:
Notice the `lock_dyn` prefix here. This is short for `lock_dynamic` which basically means that the name of the ressource we are locking is dynamic (as opposed to a static lock name such as `lock_instance_monitoring` that we use in the monitoring). We should keep this a standard in the code to make searching easier (basically the same principle as `cache_` and `batcher_`)